### PR TITLE
doctrings: make Mount and Image method args clearer

### PR DESCRIPTION
**Pull request by thundergolfer**
## Describe your changes

An initial follow-up to struggles a user had 'bringing their own `Dockerfile`. 

---

Check these boxes or delete any item (or this section) if not relevant for this PR.

- [x] Client+Server: this change is compatible with old servers
    - just docs
- [x] Client forward compatibility: this change ensures client can accept data intended for later versions of itself
    - just docs

### CODE REVIEW

File: modal/mount.py [hunk 1]
```diff
@@ -301,7 +301,8 @@ def add_local_dir(
         *,
         # Where the directory is placed within in the mount
         remote_path: Union[str, PurePosixPath, None] = None,
-        # Filter function for file selection; defaults to including all files
+        # Predicate filter function for file selection; function should return `True` for inclusion, `False` otherwise.
+        # Defaults to including all files.
```
Comment (mwaskom) [inline modal/mount.py:305]: Could we clarify that the function is passed the filepath? Maybe that's obvious but we don't explicitly say it.

**Review by mwaskom — COMMENTED** [review]
I think it would be great to also add some text in the `from_dockerfile` docstring to manage expectations — we're _translating_ Dockerfile syntax to our own Image build logic and don't currently support all commands / options. Some additional examples might be helpful too — i.e. make it more obvious that `context_mount` needs to be used if your Dockerfile has COPY commands.

**Comment by mwaskom** [comment]
Thanks for doing this by the way!

**Comment by erikbern** [comment]
Great!!

**Pull request merged**